### PR TITLE
fix : change blob size to 1gb #6302

### PR DESCRIPTION
**Pull request by jensenojs**
* change blob size to 1gb according domingozhang

## What type of PR is this?

- [ ] API-change
- [x] BUG
- [ ] Improvement
- [ ] Documentation
- [ ] Feature
- [ ] Test and CI
- [ ] Code Refactoring

## Which issue(s) this PR fixes:

issue #6302

## What this PR does / why we need it:

### CODE REVIEW

**Review by ouyuanning — APPROVED** [review]

**Comment by fengttt** [comment]
Are you sure we can handle 1G?   We said 1G before, I cannot remember when, but I am not sure we can really handle 1G.   I bet it will hit memory allocation limit somewhere in vector/mpool.

Did you try to load a 1G file?   You may need to twist mpool alloc size limit.

**Review by fengttt — CHANGES_REQUESTED** [review]
Try a 1GB file.

File: pkg/sql/plan/function/builtin/unary/load_file.go [hunk 1]
```diff
@@ -26,7 +26,8 @@ import (
 )
 
 const (
-	blobsize = 65536 // 2^16-1
+	sizeGB   = 1 << (10 * 3)
```
Comment (fengttt) [inline pkg/sql/plan/function/builtin/unary/load_file.go:29]: common has all the consts such as K/M/G defined.   Use that.

File: pkg/sql/plan/function/builtin/unary/load_file_test.go [hunk 2]
```diff
@@ -109,7 +109,7 @@ func TestLoadFile(t *testing.T) {
 	}
 
 	// Test Error
-	size := 1024 * 1024 * 1
+	size := 1<<(10*3) + 1
```
Comment (fengttt) [inline pkg/sql/plan/function/builtin/unary/load_file_test.go:112]: Why do you bother define those consts if you are not using them?  :)

Comment (jensenojs) [reply pkg/sql/plan/function/builtin/unary/load_file_test.go:112]: > Try a 1GB file.

ok

**Comment by jensenojs** [comment]
@fengttt 
Here are some problems
1. load_file will use io.readall (slow when file is big) to load data, i have tried to use io.Copy but not very helpful.
2. to solve problem below, i have tired set max-message-size for dn, cn, logservice to "2gb"
7b32c3b94c648770049ac20f245f7b9
3. after 2, there were some timeout problem and even wsl disconnection.

**Comment by jensenojs** [comment]
The current mpool does not support importing such large files. Twist mpool alloc size limit seems not a good idea according to mochen. It might be better to use s3 files instead of vectors, but dn doesn't support that right now.

**Comment by fengttt** [comment]
> @fengttt Here are some problems
> 
> 1. load_file will use io.readall (slow when file is big) to load data, i have tried to use io.Copy but not very helpful.
> 2. to solve problem below, i have tired set max-message-size for dn, cn, logservice to "2gb"
>    7b32c3b94c648770049ac20f245f7b9
> 3. after 2, there were some timeout problem and even wsl disconnection.

You cannot change rpc buffer size.   

You can change mpool size limit.   Yes, the right solution is to let CN read write S3 directly.   Whenever there need to be a "remote scan", it is scanned via a S3 locator.    

This feature need to be designed.   I don't think hacking will get you to a solution.

**Pull request closed without merging**